### PR TITLE
some code linting

### PR DESCRIPTION
**Pull request by xenohedron**
* protected constructors for abstract classes
* merge if statements
* remove unused parameters/methods
* spelling corrections
* other minor adjustments

### CODE REVIEW

File: Mage/src/main/java/mage/players/PlayerImpl.java [hunk 1]
```diff
@@ -3427,11 +3417,10 @@ public List<Permanent> getAvailableManaProducersWithCost(Game game) {
      * @param ability
      * @param availableMana if null, it won't be checked if enough mana is
      *                      available
-     * @param sourceObject
      * @param game
      * @return
      */
-    protected boolean canPlay(ActivatedAbility ability, ManaOptions availableMana, MageObject sourceObject, Game game) {
+    protected boolean canPlay(ActivatedAbility ability, ManaOptions availableMana, Game game) {
```
Comment (JayDi85) [inline Mage/src/main/java/mage/players/PlayerImpl.java:3423]: See above about source and game params removes. `sourceObject` is not used here after alternative cost fixes in #9049, but I don't like the idea to remove object's info. It's can be useful for future improvements or fixes (`ability->sourceId` and `sourceObject` are different for mdf/dfc/etc cards with multi faces, e.g. shared abilities).

shot_230715_104833

File: Mage.Server.Plugins/Mage.Player.AI.MA/src/mage/player/ai/SimulatedPlayer2.java [hunk 2]
```diff
@@ -120,7 +120,7 @@ protected void simulateOptions(Game game) {
     }
 
     @Override
-    protected void addVariableXOptions(List<Ability> options, Ability ability, int targetNum, Game game) {
```
Comment (JayDi85) [inline Mage.Server.Plugins/Mage.Player.AI.MA/src/mage/player/ai/SimulatedPlayer2.java:123]: There is interesting todo: "Support modal spells with more than one selectable mode", maybe `targetNum` needs here for that todo
shot_230715_105336

**Review by JayDi85 — APPROVED** [review]
Added some useful comments about that kind of refactoring

File: Mage/src/main/java/mage/game/permanent/PermanentImpl.java [hunk 3]
```diff
@@ -146,7 +146,7 @@ public PermanentImpl(final PermanentImpl permanent) {
         this.markedLifelink = permanent.markedLifelink;
 
         for (Map.Entry<String, List<UUID>> entry : permanent.connectedCards.entrySet()) {
-            this.connectedCards.put(entry.getKey(), entry.getValue());
+            this.connectedCards.put(entry.getKey(), new ArrayList<>(entry.getValue()));
```
Comment (xenohedron) [inline Mage/src/main/java/mage/game/permanent/PermanentImpl.java:149]: @JayDi85 just confirming this is what you mean

Comment (JayDi85) [reply Mage/src/main/java/mage/game/permanent/PermanentImpl.java:149]: yes, that's right... list value must be independent between objects

**Pull request merged**